src/apache/utilities/apache-utilities

```diff
@@ -102,3 +102,40 @@ apache_set_previous_https_port()
 {
 	snapctl set private.ports.https="$1"
 }
+
+apache_previous_compression_status()
+{
+	snapctl get private.http.compression
+}
+
+apache_set_previous_compression_status()
+{
+	snapctl set private.http.compression="$1"
+}
+
+apache_set_compression_status()
+{
+	snapctl set http.compression="$1"
+}
+
+apache_compression()
+{
+	compression="$(snapctl get http.compression)"
+
+	if [ -z "$compression" ]; then
+		compression="false"
```

Comment:
Small nit: can we pull this up to the top of the file, following the pattern established for the ports? `DEFAULT_HTTP_COMPRESSION`, perhaps.